### PR TITLE
Add a funding fee budget

### PR DESCRIPTION
**Pull request by pm47**
This is based on #2780, with a slightly different implementation.

The fee is checked right after calling `fundrawtransaction`, which makes it work for both the single and dual funding cases.

Also we use the `feeBudget` terminology instead of `maxFee`.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala [hunk 1]
```diff
@@ -205,17 +205,22 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
...
+  override def open(nodeId: PublicKey, fundingAmount: Satoshi, pushAmount_opt: Option[MilliSatoshi], channelType_opt: Option[SupportedChannelType], fundingFeerate_opt: Option[FeeratePerByte], fundingFeeBudget_opt: Option[Satoshi], announceChannel_opt: Option[Boolean], openTimeout_opt: Option[Timeout])(implicit timeout: Timeout): Future[OpenChannelResponse] = {
     // we want the open timeout to expire *before* the default ask timeout, otherwise user will get a generic response
     val openTimeout = openTimeout_opt.getOrElse(Timeout(20 seconds))
+    val fundingFeeBudget = fundingFeeBudget_opt.getOrElse{
+      val proportionalFee = Helpers.getRelayFees(appKit.nodeParams, nodeId, announceChannel_opt.getOrElse(appKit.nodeParams.channelConf.channelFlags.announceChannel)).feeProportionalMillionths.toDouble / 1e6
+      fundingAmount * proportionalFee
+    }
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala:214]: I think we should remove this fallback: if no fee budget is provided, we simply shouldn't restrict the fee. It's the node operator's responsibility to provide a sensible value here.

Comment (thomash-acinq) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala:214]: If we remove this fallback, then we should make the fee budget mandatory. It's a footgun to have a default without any limit.

Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala:214]: Good idea, making this mandatory would make sense to incentivize people to think about what value they want! We should probably also add this field to the `rbfOpen` command for consistency.

Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala:214]: How about a default budget of X% of the funding amount? So we don't have to mess with routing fees (which can be changed anyway), we don't break the api backward compat, and we remove the footgun?

Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala:214]: Why not, as long as we correctly document it! Which reminds me that this PR should update the release notes!

Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: Done in 5534091bc7, with a default of 0.1% of funding amount. It is well over default routing fee settings, but should still prevent footguns.

Another reason not to use relay fee is that users may have other reasons for opening the channel than routing. Eg. simply to send payments.

Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: Looking into `rbfOpen` now...

**Pull request merged**